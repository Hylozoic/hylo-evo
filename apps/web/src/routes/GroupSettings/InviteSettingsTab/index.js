import InviteSettingsTab from './InviteSettingsTab'
import connector from './InviteSettingsTab.connector'
export default connector(InviteSettingsTab)
