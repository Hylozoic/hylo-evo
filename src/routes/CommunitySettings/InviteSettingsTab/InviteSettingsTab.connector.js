import { connect } from 'react-redux'
import { origin } from 'util/index'
import { regenerateAccessCode, FETCH_COMMUNITY_SETTINGS } from '../CommunitySettings.store'
// import getMe from 'store/selectors/getMe'
import {
  createInvitations,
  getPendingInvites,
  expireInvitation,
  resendInvitation,
  reinviteAll
} from './InviteSettingsTab.store'

export function mapStateToProps (state, props) {
  const { community } = props
  const pending = state.pending[FETCH_COMMUNITY_SETTINGS]
  const inviteLink = origin() + community.invitePath
  const pendingInvites = getPendingInvites(state, { communityId: community.id })

  return {
    inviteLink,
    pending,
    pendingInvites
  }
}

export function mapDispatchToProps (dispatch, props) {
  const communityId = props.community.id

  return {
    regenerateAccessCode: () => dispatch(regenerateAccessCode(communityId)),
    createInvitations: (emails, message) => dispatch(createInvitations(communityId, emails, message)),
    expireInvitation: (invitationToken) => dispatch(expireInvitation(invitationToken)),
    resendInvitation: (invitationToken) => dispatch(resendInvitation(invitationToken)),
    reinviteAll: () => dispatch(reinviteAll(communityId))
  }
}

export const mergeProps = (stateProps, dispatchProps, ownProps) => {
  return {
    ...stateProps,
    ...dispatchProps,
    ...ownProps
  }
}

export default connect(mapStateToProps, mapDispatchToProps, mergeProps)
