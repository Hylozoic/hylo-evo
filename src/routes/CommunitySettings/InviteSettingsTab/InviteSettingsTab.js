import PropTypes from 'prop-types'
import React, { Component } from 'react'
import styles from './InviteSettingsTab.scss'
import Button from 'components/Button'
import Loading from 'components/Loading'
import TextareaAutosize from 'react-textarea-autosize'
import CopyToClipboard from 'react-copy-to-clipboard'
import { humanDate } from 'hylo-utils/text'
import { isEmpty } from 'lodash'
import { CSSTransitionGroup } from 'react-transition-group'

const { object, func, string } = PropTypes

const parseEmailList = emails =>
  (emails || '').split(/,|\n/).map(email => {
    var trimmed = email.trim()
    // use only the email portion of a "Joe Bloggs <joe@bloggs.org>" line
    var match = trimmed.match(/.*<(.*)>/)
    return match ? match[1] : trimmed
  })

export default class InviteSettingsTab extends Component {
  static propTypes = {
    community: object,
    regenerateAccessCode: func,
    inviteLink: string
  }

  constructor (props) {
    super(props)

    const defaultMessage = `Hi!

I'm inviting you to join ${props.community.name} community on Hylo.

${props.community.name} is using Hylo for our online community: this is our dedicated space for communication & collaboration.`

    this.state = {
      copied: false,
      reset: false,
      emails: '',
      message: defaultMessage
    }
  }

  setTemporatyState (key, value) {
    const oldValue = this.state[key]
    this.setState({[key]: value})
    setTimeout(() => {
      this.setState({[key]: oldValue})
    }, 3000)
  }

  sendInvites = () => {
    if (this.sending) return
    this.sending = true

    const { createInvitations } = this.props
    const { emails, message } = this.state

    createInvitations(parseEmailList(emails), message)
    .then(res => {
      this.sending = false
      const { invitations } = res.payload.data.createInvitation
      const badEmails = invitations.filter(email => email.error).map(e => e.email)

      const numBad = badEmails.length
      let errorMessage
      if (numBad === 1) {
        errorMessage = 'The address below is invalid.'
      } else if (numBad > 1) {
        errorMessage = `The ${numBad} addresses below are invalid.`
      }

      const numGood = invitations.length - badEmails.length
      const successMessage = numGood > 0
        ? `Sent ${numGood} ${numGood === 1 ? 'email' : 'emails'}.`
        : null

      this.setState({
        emails: badEmails.join('\n'),
        errorMessage,
        successMessage
      })
    })
  }

  render () {
    const {
      community,
      pendingCreate,
      regenerateAccessCode,
      inviteLink,
      pending,
      pendingInvites = [],
      expireInvitation,
      resendInvitation,
      reinviteAll
    } = this.props
    const { name } = community
    const { copied, reset, emails, errorMessage, successMessage } = this.state

    const onReset = () => {
      if (window.confirm("Are you sure you want to create a new join link? The current link won't work anymore if you do.")) {
        regenerateAccessCode()
        this.setTemporatyState('reset', true)
      }
    }

    const isPendingInvites = !isEmpty(pendingInvites)

    const onCopy = () => this.setTemporatyState('copied', true)

    const buttonColor = highlight => highlight ? 'green' : 'green-white-green-border'

    const disableSendBtn = (isEmpty(emails) || pendingCreate)

    const resendAllOnClick = () => {
      if (window.confirm('Are you sure you want to resend all Pending Invitations')) {
        reinviteAll()
      }
    }

    const expireOnClick = (invitationToken) => {
      expireInvitation(invitationToken)
    }

    const resendOnClick = (invitationToken) => {
      resendInvitation(invitationToken)
    }

    return <div>
      <div styleName='styles.header'>
        <div styleName='styles.title'>Invite People</div>
        <div styleName='styles.subtitle'>
          to {name} on Hylo
        </div>
      </div>
      {pending && <Loading />}
      {!pending && <div styleName='styles.invite-link-settings'>
        <div styleName='styles.invite-link-text'>
          <div styleName='styles.help'>Anyone with this link can join the community</div>
          {inviteLink && <div styleName='styles.invite-link'>{inviteLink}</div>}
          {!inviteLink && <div styleName='styles.help'>No link has been set yet</div>}
        </div>
        <div styleName='styles.buttons'>
          <Button onClick={onReset}
            color={buttonColor(reset)}
            styleName='styles.reset-button'
            narrow
            small>
            {reset
              ? 'Reset'
              : (inviteLink ? 'Reset Link' : 'New Link')}
          </Button>
          {inviteLink && <CopyToClipboard text={inviteLink} onCopy={onCopy}>
            <Button color={buttonColor(copied)} styleName='styles.copy-button' narrow small>
              {copied ? 'Copied' : 'Copy Link'}
            </Button>
          </CopyToClipboard>}
        </div>
      </div>}
      <div styleName='styles.email-section'>
        {successMessage && <span styleName='success'>{successMessage}</span>}
        {errorMessage && <span styleName='error'>{errorMessage}</span>}
        <TextareaAutosize minRows={5} styleName='styles.invite-msg-input'
          placeholder='Type email addresses'
          value={this.state.emails}
          disabled={pendingCreate}
          onChange={(event) => this.setState({emails: event.target.value})} />
        <TextareaAutosize minRows={5} styleName='styles.invite-msg-input'
          value={this.state.message}
          disabled={pendingCreate}
          onChange={(event) => this.setState({message: event.target.value})} />
        <div styleName='styles.send-invite-button'>
          <Button color='green' disabled={disableSendBtn} onClick={this.sendInvites} narrow small>
            Send Invite
          </Button>
        </div>
      </div>

      <div styleName='styles.pending-invites-section'>
        <div styleName='styles.pending-invites-header'>
          <h1 style={{flex: 1}}>{!isPendingInvites && 'No '}Pending Invites</h1>
          {isPendingInvites && <Button styleName='styles.resend-all-button'
            color='green-white-green-border'
            narrow small
            onClick={resendAllOnClick}>
            Resend All
          </Button>}
        </div>
        <div styleName='styles.pending-invites-list'>
          <CSSTransitionGroup
            transitionName={{
              enter: styles['enter'],
              enterActive: styles['enter-active'],
              leave: styles['leave'],
              leaveActive: styles['leave-active']
            }}
            transitionEnterTimeout={400}
            transitionLeaveTimeout={500}>
            {pendingInvites.map(invite => <div styleName='styles.row' key={invite.id}>
              <div style={{flex: 1}}>
                <span>{invite.email}</span>
                <span styleName='styles.invite-date'>{humanDate(invite.lastSentAt)}</span>
              </div>
              <div styleName='styles.invite-actions'>
                <span styleName='styles.action-btn styles.expire-btn' onClick={() => expireOnClick(invite.id)}>Expire</span>
                <span styleName='styles.action-btn styles.resend-btn' onClick={() => !invite.resent && resendOnClick(invite.id)}>{invite.resent ? 'Sent' : 'Resend'}</span>
              </div>
            </div>)}
          </CSSTransitionGroup>
        </div>
      </div>
    </div>
  }
}
