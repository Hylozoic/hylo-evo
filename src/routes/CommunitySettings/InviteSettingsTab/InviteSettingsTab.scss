$rowHeight: 60px;

.header {
  border-radius: 4px;
  background-color: $color-athens-gray;
  padding: $space-7x $space-6x;
  margin-bottom: $space-8x;
  margin-top: $space-6x;
}

.switch-header {
  border-radius: 4px;
  background-color: $color-athens-gray;
  padding: $space-7x $space-6x;
  margin-bottom: $space-8x;
  margin-top: $space-6x;
  display: flex;
  max-height: 80px;
}

.title {
  composes: hdr-display from 'css/typography.scss';
}

.subtitle {
  composes: caption-lt-lg from 'css/typography.scss';
}

.invite-link-settings {
  display: flex;
  align-items: flex-end;
}

.invite-link-text {
  display: flex;
  flex-direction: column;
}

.help {
  composes: caption-lt-lg from 'css/typography.scss';
  margin-bottom: 10px;
}

.invite-link {
  composes: text-button from 'css/typography.scss';
  font-size: 13px;
}

.buttons {
  margin-left: 10px;
}

.button {
  white-space: nowrap;
  padding: 0 15px;
  width: 95px;
  text-align: center;
  transition: 0.5s;
}

.reset-button {
  composes: button;
  margin-right: 10px;
}

.copy-button {
  composes: button;
  margin-top: 10px;
}

.email-section {
  display: flex;
  flex-direction: column;
  margin-top: 38px;
}

.send-invite-button {
  text-align: right;
  margin-top: 20px;
}

.email-addresses-input {
  border: 1px solid $color-ghost;
  font-size: 14px;
  height: 50px;
  border-radius: 2px;
  &::placeholder {

  }
}

.invite-msg-input {
  border: 1px solid $color-ghost;
  margin-top: 16px;
  padding: 14px;
  border-radius: 2px;
  @include font-regular;
  font-size: 14px;

  &:focus {
    outline: 0;
  }
}

.resend-all-button {
  margin-top: 3px;
}

.pending-invites-section {
  margin-top: 40px;

  .pending-invites-header {
    display: flex;
    flex-direction: row;

    h1 {
      font-size: 24px;
      color: $color-rhino-darker;
    }
  }

  .pending-invites-list {
    display: flex;
    margin-top: 10px;
    align-content: space-between;
    flex-direction: column;

    .row {
      @include font-regular;
      color: $color-rhino-60;
      font-size: 14px;
      width: 100%;
      align-items: center;
      height: $rowHeight;
      max-height: $rowHeight;
      border-top: 1px solid $color-ghost;
      display: flex;
      flex-direction: row;

      &:last-child {
        border-bottom: 1px solid $color-ghost;
      }
    }

    .invite-date {
      padding-left: 10px;
      color: #BFC5CD;
      @include font-regular;
    }

    .invite-actions {
      width: 150px;
      text-align: right;
      display: flex;
      flex-direction: row;
    }

    .action-btn {
      @include font-regular;
      cursor: pointer;
      &:hover {
        font-weight: bold;
      }
      user-select: none;
    }

    .expire-btn {
      color: $color-amaranth;
      flex: 0.5;
    }

    .resend-btn {
      color: $color-caribbean-green;
      flex: 0.5
    }
  }
}

.enter {
  opacity: 0.01;
}

.enter.enter-active {
  opacity: 1;
  transition: 400ms ease-in;
}

.leave {
  opacity: 1;
  max-height: $rowHeight;
}

.leave.leave-active {
  opacity: 0.01;
  max-height: 2px;
  transition: 500ms ease-in;
}

.error {
  color: $color-amaranth;
}

.success {
  color: $color-mountain-meadow;
}

.switch {
  float: right;
  margin-left: 180px;
  margin-top: -6px;
}

.switch-label {
  composes: caption-lt-lg from 'css/typography.scss';
}
