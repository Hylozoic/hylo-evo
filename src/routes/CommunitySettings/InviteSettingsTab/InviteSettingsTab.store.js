import { createSelector as ormCreateSelector } from 'redux-orm'
import orm from 'store/models'

export const MODULE_NAME = 'InviteSettingsTab'
export const CREATE_INVITATIONS = `${MODULE_NAME}/CREATE_INVITATIONS`
export const CREATE_INVITATIONS_PENDING = `${MODULE_NAME}/CREATE_INVITATIONS_PENDING`

export const EXPIRE_INVITATION = `${MODULE_NAME}/EXPIRE_INVITATION`
export const EXPIRE_INVITATION_PENDING = `${MODULE_NAME}/EXPIRE_INVITATION_PENDING`

export const RESEND_INVITATION = `${MODULE_NAME}/RESEND_INVITATION`
export const RESEND_INVITATION_PENDING = `${MODULE_NAME}/RESEND_INVITATION_PENDING`

export const REINVITE_ALL = `${MODULE_NAME}/REINVITE_ALL`
export const REINVITE_ALL_PENDING = `${MODULE_NAME}/REINVITE_ALL_PENDING`

const defaultState = []

export default function reducer (state = defaultState, action) {
  const { error, type } = action
  if (error) return state

  switch (type) {
    default:
      return state
  }
}

export function createInvitations (communityId, emails, message) {
  return {
    type: CREATE_INVITATIONS,
    graphql: {
      query: `mutation ($communityId: ID, $data: InviteInput) {
        createInvitation(communityId: $communityId, data: $data) {
          invitations {
            id,
            email,
            created_at
            last_sent_at
          }
        }
      }`,
      variables: {
        communityId,
        data: {
          emails,
          message
        }
      }
    },
    meta: {
      communityId,
      emails,
      optimistic: true
    }
  }
}

export function reinviteAll (communityId) {
  return {
    type: REINVITE_ALL,
    graphql: {
      query: `mutation ($communityId: ID) {
        reinviteAll(communityId: $communityId) {
          success
        }
      }`,
      variables: {
        communityId
      }
    },
    meta: {
      communityId,
      optimistic: true
    }
  }
}

export function expireInvitation (invitationToken) {
  return {
    type: EXPIRE_INVITATION,
    graphql: {
      query: `mutation ($invitationToken: ID) {
        expireInvitation(invitationToken: $invitationToken) {
          success
        }
      }`,
      variables: {
        invitationToken
      }
    },
    meta: {
      invitationToken,
      optimistic: true
    }
  }
}

export function resendInvitation (invitationToken) {
  return {
    type: RESEND_INVITATION,
    graphql: {
      query: `mutation ($invitationToken: ID) {
        resendInvitation(invitationToken: $invitationToken) {
          success
        }
      }`,
      variables: {
        invitationToken
      }
    },
    meta: {
      invitationToken,
      optimistic: true
    }
  }
}

// expects props to be of the form {communityId}
export const getPendingInvites = ormCreateSelector(
  orm,
  state => state.orm,
  (state, props) => props.communityId,
  ({ Community }, id) => {
    const community = Community.safeGet({id})
    if (!community || !community.pendingInvitations) return []
    community.pendingInvitations.orderBy(i => -new Date(i.created_at)).toModelArray()
    return community
    .pendingInvitations
    .orderBy(i => -new Date(i.created_at))
    .filter(invite => invite.id)
    .toModelArray()
  }
)
