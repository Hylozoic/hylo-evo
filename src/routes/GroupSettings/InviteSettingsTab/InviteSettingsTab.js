import PropTypes from 'prop-types'
import React, { Component } from 'react'
import { withTranslation } from 'react-i18next'
import Button from 'components/Button'
import Loading from 'components/Loading'
import TextareaAutosize from 'react-textarea-autosize'
import CopyToClipboard from 'react-copy-to-clipboard'
import { TextHelpers } from 'hylo-shared'
import { isEmpty } from 'lodash'
import { TransitionGroup, CSSTransition } from 'react-transition-group'
import Icon from 'components/Icon'
import isMobile from 'ismobilejs'
import ReactTooltip from 'react-tooltip'

import styles from './InviteSettingsTab.scss'

const { object, func, string } = PropTypes

const parseEmailList = emails =>
  (emails || '').split(/,|\n/).map(email => {
    const trimmed = email.trim()
    // use only the email portion of a "Joe Bloggs <joe@bloggs.org>" line
    const match = trimmed.match(/.*<(.*)>/)
    return match ? match[1] : trimmed
  })

class InviteSettingsTab extends Component {
  static propTypes = {
    group: object,
    regenerateAccessCode: func,
    inviteLink: string
  }

  constructor (props) {
    super(props)

    const defaultMessage = this.props.t(`Hi!

I'm inviting you to join {{name}} on Hylo.

{{name}} is using Hylo for our online community: this is our dedicated space for communication & collaboration.1`, { name: props.group.name })
    this.state = {
      copied: false,
      reset: false,
      emails: '',
      message: defaultMessage
    }
  }

  setTemporatyState (key, value) {
    const oldValue = this.state[key]
    this.setState({ [key]: value })
    setTimeout(() => {
      this.setState({ [key]: oldValue })
    }, 3000)
  }

  handleSendInvites = () => {
    if (this.sending) return
    this.sending = true

    const { createInvitations, trackAnalyticsEvent, t } = this.props
    const { emails, message } = this.state

    createInvitations(parseEmailList(emails), message)
      .then(res => {
        this.sending = false
        const { invitations } = res.payload.data.createInvitation
        const badEmails = invitations.filter(email => email.error).map(e => e.email)

        const numBad = badEmails.length
        let errorMessage, successMessage
        if (numBad > 0) {
          errorMessage = `${t('{{numBad}} invalid email address/es found (see above)).', { numBad })}{' '}`
        }
        const numGood = invitations.length - badEmails.length
        if (numGood > 0) {
          successMessage = t(`Sent {{numGood}} {{email}}`, { numGood, email: numGood === 1 ? 'email' : 'emails' })
          trackAnalyticsEvent('Group Invitations Sent', { numGood })
        }
        this.setState({
          emails: badEmails.join('\n'),
          errorMessage,
          successMessage
        })
      })
  }

  render () {
    const {
      group,
      pendingCreate,
      regenerateAccessCode,
      inviteLink,
      pending,
      pendingInvites = [],
      expireInvitation,
      resendInvitation,
      reinviteAll,
      t
    } = this.props
    const { copied, reset, emails, errorMessage, successMessage } = this.state

    const onReset = () => {
      if (window.confirm(t("Are you sure you want to create a new join link? The current link won't work anymore if you do."))) {
        regenerateAccessCode()
        this.setTemporatyState('reset', true)
      }
    }

    const hasPendingInvites = !isEmpty(pendingInvites)

    const onCopy = () => this.setTemporatyState('copied', true)

    const buttonColor = highlight => highlight ? 'green' : 'green-white-green-border'

    const disableSendBtn = (isEmpty(emails) || pendingCreate)

    const resendAllOnClick = () => {
      if (window.confirm(t('Are you sure you want to resend all Pending Invitations'))) {
        reinviteAll()
      }
    }

    const expireOnClick = (invitationToken) => {
      expireInvitation(invitationToken)
    }

    const resendOnClick = (invitationToken) => {
      resendInvitation(invitationToken)
    }

    return (
      <div styleName='styles.container'>
        <div styleName='styles.header'>
          <div styleName='styles.title'>{t('Invite People')}</div>
        </div>

        {pending && <Loading />}

        {!pending && (
          <>
            <div styleName='styles.invite-link-section'>
              <div styleName='styles.subtitle'>
                {t('Share a Join Link')}
              </div>
              <div styleName='styles.help'>
                {t('Anyone can join ')}<span style={{ fontWeight: 'bold' }}>{group.name}</span> {t('with this link')}.{' '}{inviteLink && t('Click or press on it to copy it')}:
              </div>
              <div styleName='styles.invite-link-settings'>
                {inviteLink && (
                  <div styleName='styles.invite-link'>
                    {!copied && (
                      <>
                        <CopyToClipboard text={inviteLink} onCopy={onCopy}>
                          <span data-tip={t('Click to Copy')} data-for='invite-link-tooltip'>
                            {inviteLink}
                            <Icon name='Copy' styleName='styles.copy-icon' />
                          </span>
                        </CopyToClipboard>
                        {!isMobile.any && (
                          <ReactTooltip
                            place='top'
                            type='dark'
                            id='invite-link-tooltip'
                            effect='solid'
                            delayShow={500}
                          />
                        )}
                      </>
                    )}
                    {copied && t('Copied!')}
                  </div>
                )}
                <Button onClick={onReset} styleName='styles.invite-link-button' color={buttonColor(reset)}>
                  {inviteLink ? t('Reset Link') : t('Generate a Link')}
                </Button>
              </div>
            </div>
          </>
        )}

        <div styleName='styles.email-section'>
          <div styleName='styles.subtitle'>
            {t('Send Invites via email')}
          </div>
          <div styleName='styles.help'>{t('Email addresses of those you\'d like to invite:')}</div>
          <TextareaAutosize
            minRows={1}
            styleName='styles.invite-msg-input'
            placeholder={t('Type email addresses (multiples should be separated by either a comma or new line)')}
            value={this.state.emails}
            disabled={pendingCreate}
            onChange={(event) => this.setState({ emails: event.target.value })}
          />
          <div styleName='styles.help'>{t('Customize the invite email message (optional):')}</div>
          <TextareaAutosize
            minRows={5}
            styleName='styles.invite-msg-input'
            value={this.state.message}
            disabled={pendingCreate}
            onChange={(event) => this.setState({ message: event.target.value })}
          />
          <div styleName='styles.send-invite-button'>
            <div styleName='styles.send-invite-feedback'>
              {errorMessage && <span styleName='error'>{errorMessage}</span>}
              {successMessage && <span styleName='success'>{successMessage}</span>}
            </div>
            <Button color='green' disabled={disableSendBtn} onClick={this.handleSendInvites} narrow small>
              {t('Send Invite')}
            </Button>
          </div>
        </div>

        {hasPendingInvites && (
          <div styleName='styles.pending-invites-section'>
            <div styleName='styles.pending-invites-header'>
              <div styleName='styles.subtitle'>{t('Pending Invites')}</div>
              {hasPendingInvites && (
                <Button
                  styleName='styles.resend-all-button'
                  color='green-white-green-border'
                  narrow small
                  onClick={resendAllOnClick}
                >
                  {t('Resend All')}
                </Button>
              )}
            </div>
            <div styleName='styles.pending-invites-list'>
              <TransitionGroup>
                {pendingInvites.map((invite, index) => (
                  <CSSTransition
                    classNames={{
                      enter: styles.enter,
                      enterActive: styles['enter-active'],
                      exit: styles.exit,
                      exitActive: styles['exit-active']
                    }}
                    timeout={{ enter: 400, exit: 500 }}
                    key={index}
                  >
                    <div styleName='styles.row' key={invite.id}>
                      <div style={{ flex: 1 }}>
                        <span>{invite.email}</span>
                        <span styleName='styles.invite-date'>{TextHelpers.humanDate(invite.lastSentAt)}</span>
                      </div>
                      <div styleName='styles.invite-actions'>
                        <span styleName='styles.action-btn styles.expire-btn' onClick={() => expireOnClick(invite.id)}>{t('Expire')}</span>
                        <span styleName='styles.action-btn styles.resend-btn' onClick={() => !invite.resent && resendOnClick(invite.id)}>{invite.resent ? this.props('Sent') : this.props('Resend')}</span>
                      </div>
                    </div>
                  </CSSTransition>
                ))}
              </TransitionGroup>
            </div>
          </div>
        )}
      </div>
    )
  }
}
export default withTranslation()(InviteSettingsTab)
