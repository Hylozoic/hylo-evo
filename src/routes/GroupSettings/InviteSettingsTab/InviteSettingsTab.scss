$rowHeight: 60px;

.container {
  // empty
}

.header {
  border-radius: 4px;
  background-color: $color-athens-gray;
  margin-bottom: $space-2x;
}

.title {
  composes: hdr-display from 'css/typography.scss';
}

.subtitle {
  composes: hdr-headline from 'css/typography.scss';
  color: $color-rhino-80;
  margin-bottom: 10px;
  padding: 0
}

.help {
  color: $color-rhino-60;
  margin-bottom: 10px;
}

.invite-link-section {
    display: flex;
    flex-direction: column;
    margin-bottom: 26px;
}

.invite-link-settings {
  display: flex;
  flex-direction: column;
}

.invite-link {
  // composes: text-button from 'css/typography.scss';
  font-size: 13px;
  flex-wrap: wrap;
  color: $color-caribbean-green;
  cursor: pointer;
  margin-bottom: 10px
}

.button {
  height: max-content;
  width: max-content;
  line-height: initial;
  font-size: 12px;
  color: $color-caribbean-green;
  padding: 5px 10px 5px 10px;
  margin: 0;
  &:hover {
    text-decoration: none;
    color: $color-gossamer;
  }
}

.invite-link-button {
  composes: button;
}

.copy-icon {
  font-size: 12px;
  margin-left: 5px;
  margin-right: 10px
}

.email-section {
  display: flex;
  flex-direction: column;
  margin-bottom: 26px;
}

.email-addresses-input {
  border: 1px solid $color-ghost;
  font-size: 14px;
  height: 50px;
  border-radius: 2px;
  margin-bottom: 16px;
  &::placeholder {
  }
}

.invite-msg-input {
  border: 1px solid $color-ghost;
  padding: 10px;
  border-radius: 2px;
  @include font-regular;
  font-size: 14px;
  margin-bottom: 16px;
  &:focus {
    outline: 0;
  }
}

.send-invite-button {
  display: flex;
  justify-content: space-between;
}

.resend-all-button {
  composes: button;
  margin-bottom: 10px;
}

.pending-invites-section {
  .pending-invites-header {
    display: flex;
    align-items: flex-end;
    justify-content: space-between;
  }

  .pending-invites-list {
    display: flex;
    align-content: space-between;
    flex-direction: column;

    .row {
      @include font-regular;
      color: $color-rhino-60;
      font-size: 14px;
      width: 100%;
      align-items: center;
      height: $rowHeight;
      max-height: $rowHeight;
      border-top: 1px solid $color-ghost;
      display: flex;
      flex-direction: row;

      &:last-child {
        border-bottom: 1px solid $color-ghost;
      }
    }

    .invite-date {
      padding-left: 10px;
      color: #BFC5CD;
      @include font-regular;
    }

    .invite-actions {
      width: 150px;
      text-align: right;
      display: flex;
      flex-direction: row;
    }

    .action-btn {
      @include font-regular;
      cursor: pointer;
      &:hover {
        font-weight: bold;
      }
      user-select: none;
    }

    .expire-btn {
      color: $color-amaranth;
      flex: 0.5;
    }

    .resend-btn {
      color: $color-caribbean-green;
      flex: 0.5
    }
  }
}

.enter {
  opacity: 0.01;
}

.enter.enter-active {
  opacity: 1;
  transition: 400ms ease-in;
}

.leave {
  opacity: 1;
  max-height: $rowHeight;
}

.leave.leave-active {
  opacity: 0.01;
  max-height: 2px;
  transition: 500ms ease-in;
}

.error {
  color: $color-amaranth;
}

.success {
  color: $color-caribbean-green;
}

@media screen and (max-width: 525px) {
  .invite-link-settings {
    // display: block;
    // text-align: center;
  }
}