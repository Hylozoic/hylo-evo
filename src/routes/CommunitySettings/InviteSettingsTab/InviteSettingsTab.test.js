import InviteSettingsTab from './InviteSettingsTab'
import { shallow } from 'enzyme'
import React from 'react'

it('renders correctly', () => {
  const community = {
    id: 1,
    name: 'Hylo'
  }

  const wrapper = shallow(<InviteSettingsTab
    community={community}
    regenerateAccessCode={() => {}}
    inviteLink='http://www.hylo.com/c/hylo/join/lalala'
    />)
  expect(wrapper).toMatchSnapshot()
})

it('toggles allowCommunityInvites and calls the function to make the request on the server', () => {
  const community = {
    id: 1,
    name: 'Hylo'
  }
  const allMembersCanInvite = false
  const allowCommunityInvites = jest.fn(() => new Promise(() => {}))

  const wrapper = shallow(<InviteSettingsTab
    community={community}
    regenerateAccessCode={() => {}}
    inviteLink='http://www.hylo.com/c/hylo/join/lalala'
    allMembersCanInvite={allMembersCanInvite}
    allowCommunityInvites={allowCommunityInvites}
    />)
  wrapper.instance().toggleSwitch()
  expect(allowCommunityInvites).toBeCalled()
  expect(wrapper.instance().state.allMembersCanInvite).toBeTruthy()
})
