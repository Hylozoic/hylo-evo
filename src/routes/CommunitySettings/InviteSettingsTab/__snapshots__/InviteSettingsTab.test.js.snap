// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    data-stylename="styles.header"
  >
    <div
      data-stylename="styles.title"
    >
      Invite People
    </div>
    <div
      data-stylename="styles.subtitle"
    >
      to 
      Hylo
       on Hylo
    </div>
  </div>
  <div
    data-stylename="styles.invite-link-settings"
  >
    <div
      data-stylename="styles.invite-link-text"
    >
      <div
        data-stylename="styles.help"
      >
        Anyone with this link can join the community
      </div>
      <div
        data-stylename="styles.invite-link"
      >
        http://www.hylo.com/c/hylo/join/lalala
      </div>
    </div>
    <div
      data-stylename="styles.buttons"
    >
      <Button
        color="green-white-green-border"
        data-stylename="styles.reset-button"
        narrow={true}
        onClick={[Function]}
        small={true}
      >
        Reset Link
      </Button>
      <CopyToClipboard
        onCopy={[Function]}
        text="http://www.hylo.com/c/hylo/join/lalala"
      >
        <Button
          color="green-white-green-border"
          data-stylename="styles.copy-button"
          narrow={true}
          small={true}
        >
          Copy Link
        </Button>
      </CopyToClipboard>
    </div>
  </div>
  <div
    data-stylename="styles.email-section"
  >
    <TextareaAutosize
      data-stylename="styles.invite-msg-input"
      minRows={5}
      onChange={[Function]}
      onHeightChange={[Function]}
      placeholder="Type email addresses"
      useCacheForDOMMeasurements={false}
      value=""
    />
    <TextareaAutosize
      data-stylename="styles.invite-msg-input"
      minRows={5}
      onChange={[Function]}
      onHeightChange={[Function]}
      useCacheForDOMMeasurements={false}
      value="Hey! Hereâ€™s an invite to the Hylo community on Hylo."
    />
    <div
      data-stylename="styles.send-invite-button"
    >
      <Button
        color="green"
        disabled={true}
        narrow={true}
        onClick={[Function]}
        small={true}
      >
        Send Invite
      </Button>
    </div>
  </div>
  <div
    data-stylename="styles.pending-invites-section"
  >
    <div
      data-stylename="styles.pending-invites-header"
    >
      <h1
        style={
          Object {
            "flex": 1,
          }
        }
      >
        No 
        Pending Invites
      </h1>
    </div>
    <div
      data-stylename="styles.pending-invites-list"
    >
      <CSSTransitionGroup
        transitionAppear={false}
        transitionEnter={true}
        transitionEnterTimeout={400}
        transitionLeave={true}
        transitionLeaveTimeout={500}
        transitionName={
          Object {
            "enter": undefined,
            "enterActive": undefined,
            "leave": undefined,
            "leaveActive": undefined,
          }
        }
      />
    </div>
  </div>
</div>
`;
